SPECS/glusterfs/glusterfs.spec

```diff
@@ -34,463 +45,666 @@
 %else
 %global _with_tmpfilesdir --without-tmpfilesdir
 %endif
 # without server should also disable some server-only components
-%if ( 0%{?_without_server:1} )
+%if 0%{?_without_server:1}
 %global _without_events --disable-events
 %global _without_georeplication --disable-georeplication
+%global _without_linux_io_uring --disable-linux-io_uring
 %global _with_gnfs %{nil}
 %global _without_ocf --without-ocf
 %endif
+%if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} > 7 ) || ( 0%{?mariner} && 0%{?mariner} >= 3 )
+%global _usepython3 1
 %global _pythonver 3
-%global service_start()   /bin/systemctl --quiet start %{1}.service || : \
+%else
+%global _usepython3 0
+%global _pythonver 2
+%endif
+# From https://fedoraproject.org/wiki/Packaging:Python#Macros
+%if ( 0%{?rhel} && 0%{?rhel} <= 6 ) && ( 0%{!?mariner} )
+%{!?python2_sitelib: %global python2_sitelib %(python2 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")}
+%{!?python2_sitearch: %global python2_sitearch %(python2 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(1))")}
+%global _rundir %{_localstatedir}/run
+%endif
+%if ( 0%{?_with_systemd:1} )
+%global service_start()   /bin/systemctl --quiet start %1.service || : \
 %{nil}
-%global service_stop()    /bin/systemctl --quiet stop %{1}.service || :\
+%global service_stop()    /bin/systemctl --quiet stop %1.service || :\
 %{nil}
-%global service_install() install -D -p -m 0644 %{_sourcedir}/%{1}.service %{buildroot}%{2} \
+%global service_install() install -D -p -m 0644 %1.service %{buildroot}%2 \
 %{nil}
 # can't seem to make a generic macro that works
 %global glusterd_svcfile   %{_unitdir}/glusterd.service
 %global glusterfsd_svcfile %{_unitdir}/glusterfsd.service
 %global glusterta_svcfile %{_unitdir}/gluster-ta-volume.service
 %global glustereventsd_svcfile %{_unitdir}/glustereventsd.service
 %global glusterfssharedstorage_svcfile %{_unitdir}/glusterfssharedstorage.service
+%else
+%global systemd_post()  /sbin/chkconfig --add %1 >/dev/null 2>&1 || : \
+%{nil}
+%global systemd_preun() /sbin/chkconfig --del %1 >/dev/null 2>&1 || : \
+%{nil}
+%global systemd_postun_with_restart() /sbin/service %1 condrestart >/dev/null 2>&1 || : \
+%{nil}
+%global service_start()   /sbin/service %1 start >/dev/null 2>&1 || : \
+%{nil}
+%global service_stop()    /sbin/service %1 stop >/dev/null 2>&1 || : \
+%{nil}
+%global service_install() install -D -p -m 0755 %1.init %{buildroot}%2 \
+%{nil}
+# can't seem to make a generic macro that works
+%global glusterd_svcfile   %{_sysconfdir}/init.d/glusterd
+%global glusterfsd_svcfile %{_sysconfdir}/init.d/glusterfsd
+%global glustereventsd_svcfile %{_sysconfdir}/init.d/glustereventsd
+%endif
 %{!?_pkgdocdir: %global _pkgdocdir %{_docdir}/%{name}-%{version}}
+# We do not want to generate useless provides and requires for xlator
+# .so files to be set for glusterfs packages.
+# Filter all generated:
+#
+# TODO: RHEL5 does not have a convenient solution
```

Comment:
Please also remove comments referring to other distros, since they don't make much sense here. Ideally, also remove the ones, which are still here from the 2.0 version, but I'm not going to push for that.:)

Reply:
i started a thread to come to some understanding about modifying upstream bits (like this spec file).

Reply:
removed